### ISSUE TITLE
Build my own model and add it to the models in your package

### ISSUE THREAD

**Issue by fahmanali** [creafz/pytorch-cnn-finetune#11]
If I want to build my model and benefit from your code capability to learn or predict images with different  sizes

**Comment by creafz** [comment]
The library was built mainly to work with existing models from other packages.

You can use the procedure below to add your own model, but I am not sure that it will work correctly with all edge cases.
1. Import required modules:
```
import torch
import torch.nn as nn

from cnn_finetune import make_model
from cnn_finetune.base import ModelWrapperBase
from cnn_finetune.shims import no_grad_variable
from cnn_finetune.utils import product
```
2. Make your own model, but without a global pooling layer, without dropout, and without a classifier, for example:
```
class MyCnn(nn.Module):

    def __init__(self):
        super(MyCnn, self).__init__()
        self.layer1 = nn.Sequential(
            nn.Conv2d(3, 16, kernel_size=5, stride=1, padding=2),
            nn.BatchNorm2d(16),
            nn.ReLU(),
            nn.MaxPool2d(kernel_size=2, stride=2))

        self.layer2 = nn.Sequential(
            nn.Conv2d(16, 32, kernel_size=5, stride=1, padding=2),
            nn.BatchNorm2d(32),
            nn.ReLU(),
            nn.MaxPool2d(kernel_size=2, stride=2),
        )

    def forward(self, x):
        out = self.layer1(x)
        out = self.layer2(out)
        return out
```


3. Make a wrapper for the model above:
```
class MyCnnWrapper(ModelWrapperBase):
    
    # This model name will be used in a `make_model` call.
    # You can use any unique identifier.
    model_names = ['mycnn']
   
    def get_original_model(self):
        return MyCnn()

    def get_features(self, original_model):
        return original_model

    def get_classifier_in_features(self, original_model):
        # This method returns the number of input features for classifier.
        input_size = self.input_size or (224, 224)
        with no_grad_variable(torch.zeros(1, 3, *input_size)) as input_var:
            self.eval()
            output = self.features(input_var)
            if self.pool is not None:
                output = self.pool(output)
            self.train()
            return product(output.size()[1:])
```

4. Make a new model with `cnn_finetune`. 

```model = make_model('mycnn', num_classes=19)```

Note that if the definition of `MyCnnWrapper` is located in another module you first need to import `MyCnnWrapper`  so it will register a model with the name `mycnn` in a global model registry.
```
from cnn_finetune import make_model
from .some_module import MyCnnWrapper

model = make_model('mycnn', num_classes=19)
```
